iree/tools/opt_main.cc

```diff
@@ -0,0 +1,149 @@
+void registerSymbolTestPasses();
+void registerTestAffineDataCopyPass();
+void registerTestAllReduceLoweringPass();
+void registerTestCallGraphPass();
+void registerTestConstantFold();
+void registerTestFunc();
+void registerTestGpuMemoryPromotionPass();
+void registerTestLinalgTransforms();
+void registerTestLivenessPass();
+void registerTestLoopFusion();
+void registerTestLoopMappingPass();
+void registerTestMatchers();
+void registerTestMemRefDependenceCheck();
+void registerTestMemRefStrideCalculation();
+void registerTestOpaqueLoc();
+void registerTestParallelismDetection();
+void registerTestVectorConversions();
+void registerTestVectorToLoopsPass();
+void registerVectorizerTestPass();
+}  // namespace mlir
+
+static llvm::cl::opt<std::string> inputFilename(llvm::cl::Positional,
+                                                llvm::cl::desc("<input file>"),
+                                                llvm::cl::init("-"));
+
+static llvm::cl::opt<std::string> outputFilename(
+    "o", llvm::cl::desc("Output filename"), llvm::cl::value_desc("filename"),
+    llvm::cl::init("-"));
+
+static llvm::cl::opt<bool> splitInputFile(
+    "split-input-file",
+    llvm::cl::desc("Split the input file into pieces and process each "
+                   "chunk independently"),
+    llvm::cl::init(false));
+
+static llvm::cl::opt<bool> verifyDiagnostics(
+    "verify-diagnostics",
+    llvm::cl::desc("Check that emitted diagnostics match "
+                   "expected-* lines on the corresponding line"),
+    llvm::cl::init(false));
+
+static llvm::cl::opt<bool> verifyPasses(
+    "verify-each",
+    llvm::cl::desc("Run the verifier after each transformation pass"),
+    llvm::cl::init(true));
+
+namespace mlir {
+void registerTestPasses() {
+  mlir::registerConvertToTargetEnvPass();
+  mlir::registerInliner();
+  mlir::registerMemRefBoundCheck();
+  mlir::registerPassManagerTestPass();
+  mlir::registerPatternsTestPass();
+  mlir::registerPrintOpAvailabilityPass();
+  mlir::registerSimpleParametricTilingPass();
+  mlir::registerSymbolTestPasses();
+  mlir::registerTestAffineDataCopyPass();
+  mlir::registerTestAllReduceLoweringPass();
+  mlir::registerTestCallGraphPass();
+  mlir::registerTestConstantFold();
+  mlir::registerTestFunc();
+  mlir::registerTestGpuMemoryPromotionPass();
+  mlir::registerTestLinalgTransforms();
+  mlir::registerTestLivenessPass();
+  mlir::registerTestLoopFusion();
+  mlir::registerTestLoopMappingPass();
+  mlir::registerTestMatchers();
+  mlir::registerTestMemRefDependenceCheck();
+  mlir::registerTestMemRefStrideCalculation();
+  mlir::registerTestOpaqueLoc();
+  mlir::registerTestParallelismDetection();
+  mlir::registerTestVectorConversions();
+  mlir::registerTestVectorToLoopsPass();
+  mlir::registerVectorizerTestPass();
+
+  // The following passes are using global initializers, just link them in.
+  if (std::getenv("bar") != (char *)-1) return;
+
+  // TODO: move these to the test folder.
```

Comment:
I think this TODO doesn't apply to us. It should be TODO delete these :-D

Reply:
DITO (#880).